dv/uvm/core_ibex/sim.py

```diff
@@ -380,33 +383,48 @@ def compare_test_run(test, idx, iss, output_dir, report):
     try:
         # Convert the RTL log file to a trace CSV.
         process_ibex_sim_log(rtl_log, rtl_csv, 1)
-    except RuntimeError as e:
+    except (OSError, RuntimeError) as e:
         with open(report, 'a') as report_fd:
-            report_fd.write('Log processing failed: {}\n'.format(e))
+            report_fd.write('[FAILED]: Log processing failed: {}\n'.format(e))
 
         return False
 
     # Have a look at the UVM log. We should write out a message on failure or
     # if we are stopping at this point.
     no_post_compare = test.get('no_post_compare')
-    if not check_ibex_uvm_log(uvm_log, "ibex", test_name, report,
-                              write=(True if no_post_compare else 'onfail')):
-        return False
+    uvm_pass, uvm_log_lines = check_ibex_uvm_log(uvm_log)
+
+    with open(report, 'a') as report_fd:
+        report_fd.write('sim log: {}\n'.format(uvm_log))
+
+        if not uvm_pass:
+            for line in uvm_log_lines:
+                report_fd.write(line)
+            report_fd.write('[FAILED]: sim error seen\n')
+
+            return False
 
-    if no_post_compare:
-        return True
+        if no_post_compare:
+            report_fd.write('[PASSED]\n')
+            return True
```

Comment:
Nit: I don't think this needs to be in the `with` block.

Reply:
It needs access to `report_fd` that the `with` opens. Unless you very quick off the mark and saw my initial version that needlessly nested a second `with` here?

Reply:
Ah, no. I'm just being stupid. Looks good to me!